doc/how-to-run-in-windows.rst

```diff
@@ -0,0 +1,352 @@
+``openssl``, one for `msys <https://packages.msys2.org/base/openssl>`_ and one
+for `every other environment
+<https://packages.msys2.org/base/mingw-w64-openssl>`_ (and in particular for
+CLANG64 or UCRT64). When installing packages from MSYS2, one should pick the
+options that show ``ucrt64/`` or ``clang64/`` as prefixes, and avoid the ones
+that show ``msys/``.
+
+::
+
+   msys2> pacman -Ss openssl
+   ucrt64/mingw-w64-ucrt-x86_64-openssl 3.2.0-1
+       The Open Source toolkit for Secure Sockets Layer and Transport Layer Security (mingw-w64)
+   ...
+   clang64/mingw-w64-clang-x86_64-openssl 3.2.0-1
+       The Open Source toolkit for Secure Sockets Layer and Transport Layer Security (mingw-w64)
+   ...
+   msys/openssl 3.2.0-1
+       The Open Source toolkit for Secure Sockets Layer and Transport Layer Security
+
+   msys2> pacman -S ucrt64/mingw-w64-ucrt-x86_64-openssl
+
+Once the package is installed, the appropriate location of the libraries should
+be given to GHC, for example if compiling something that depends on
+``libssl.a``:
+
+::
+
+   > ghc ... -IC:\\msys64\\ucrt64\\includes -LC:\\msys64\\ucrt64\\lib -lssl
+
+This should result in a succesful compilation.
+
+Using UNIX-like binaries with GHC
+---------------------------------
+
+As said above, the binaries that we might need to use are (by default) located
+in two places: ``C:\msys64\usr\bin`` and ``C:\msys64\<environment>\bin``. The
+easiest way to give GHC access to these executables is by including those
+directories in your ``PATH``.
+
+- On PowerShell:
+  ::
+
+     PS> $env:Path += ';C:\msys64\usr\bin;C:\msys64\ucrt64\bin'
+
+     PS> (Get-Command make.exe).Path
+     C:\msys64\usr\bin\make.exe
+
+     PS> make --version
+     GNU Make 4.4.1
+     Built for x86_64-pc-msys
+     ...
+
+- On ``cmd``:
+  ::
+
+     cmd> set PATH=%PATH%;C:\msys64\usr\bin;C:\msys64\ucrt64\bin
+
+     cmd> where make
+     C:\msys64\usr\bin\make.exe
+
+     cmd> make --version
+     GNU Make 4.4.1
+     Built for x86_64-pc-msys
+     ...
+
+- On MSYS2 this is usually already set (otherwise your whole system is probably
+  broken), but just in case one would do the following:
+
+  ::
+
+     msys2> export PATH=$PATH:/c/msys64/usr/bin:/c/msys64/ucrt64/bin
+
+     msys2> which make
+     /usr/bin/make
+
+     msys2> make --version
+     GNU Make 4.4.1
+     Built for x86_64-pc-msys
+     ...
+
+If one really wants, it is possible to edit the system ``PATH`` to add these two
+directories (although as explained below, Cabal will automatically add them for
+us) by doing:
+
+1. Open the search bar and search for "Edit the system environment variables"
+
+2. Switch to the "Advanced" tab and click on "Environment Variables..."
+
+3. Double click on ``PATH`` and add ``;<dir1>;<dir2>`` at the end of the already existing value
+
+Once these directories are in the path, any UNIX-like utilities we might need
+will be in scope, so configuring packages should be able to find them.
+
+Using Cabal on Windows

```

Comment:
Should this section swap with the one immediately above? I feel like `cabal` is the standard way most people are going to trigger GHC.